app/src/main/java/org/connectbot/transport/SSH.java

```diff
@@ -108,7 +108,7 @@ public SSH(HostBean host, TerminalBridge bridge, TerminalManager manager) {
 	private final static int AUTH_TRIES = 20;
 
 	private static final Pattern hostmask = Pattern.compile(
-			"^(.+)@(([0-9a-z.-]+)|(\\[[a-f:0-9]+\\]))(:(\\d+))?$", Pattern.CASE_INSENSITIVE);
+			"^(.+)@(([0-9a-z.-_]+)|(\\[[a-f:0-9]+\\]))(:(\\d+))?$", Pattern.CASE_INSENSITIVE);
```

Comment:
The hyphen must be the last in the regex or it changes the meaning to "all characters starting with . (0x2e) and extending to _ (0x5f)".

Reply:
Done.